src/Umbraco.Web/PropertyEditors/NestedContentPropertyEditor.cs

```diff
@@ -266,6 +266,45 @@ public override object FromEditor(ContentPropertyData editorValue, object curren
                 return JsonConvert.SerializeObject(value);
             }
 
+            public IEnumerable<UmbracoEntityReference> GetReferences(object value)
+            {
+                var rawJson = value == null ? string.Empty : value is string str ? str : value.ToString();
+
+                var result = new List<UmbracoEntityReference>();
+
+                var list = JsonConvert.DeserializeObject<List<object>>(rawJson);
+                if (list == null)
+                    return result;
+
+                foreach (var o in list)
+                {
+                    var propValues = (JObject) o;
+
+                    var contentType = GetElementType(propValues);
```

Comment:
This is an N+1 call, but I realize that this is being done in the `ConvertDbToString` method too. I also realize that all content types are cached, so it's not hitting the DB each time but still something to ponder ;)

Reply:
On that note, this code is basically all duplicated from the ConvertDbToString method. We should prob make a re-usable method to do this iteration

Reply:
Actually this was also what I did first and I ended up in something similar to what you did, but I was not proud of the readability of the delegate similar to this: `Action<string, PropertyType, JObject, int> onIteration`. 😢 

But the code duplication was not good either.

Reply:
Yeah, it's not the prettiest but I'll take it over the duplication, at least its isolated to this editor.